presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergSplitManager.java

```diff
@@ -179,32 +186,63 @@ public void testSplitSchedulingWithTablePropertyAndSession()
                 Long.toString(TableProperties.SPLIT_SIZE_DEFAULT));
         assertQuerySucceeds("ALTER TABLE test_split_size SET PROPERTIES (\"read.split.target-size\" = 1)");
         String selectQuery = "SELECT * FROM test_split_size";
-        long maxSplits = getSplitsForSql(session, selectQuery);
+        long maxSplits = getSplitsForSql(session, selectQuery).size();
 
         IntStream.range(1, 5)
                 .mapToObj(i -> Math.pow(2, i))
                 .forEach(splitSize -> {
                     assertQuerySucceeds("ALTER TABLE test_split_size SET PROPERTIES (\"read.split.target-size\" =" + splitSize.intValue() + ")");
-                    assertEquals(getSplitsForSql(session, selectQuery), (double) maxSplits / splitSize, 5);
+                    assertEquals(getSplitsForSql(session, selectQuery).size(), (double) maxSplits / splitSize, 5);
                 });
         // split size should be set to 32 on the table property.
         // Set it to 1 with the session property to override the table value and verify we get the
         // same number of splits as when the table value is set to 1.
         Session minSplitSession = Session.builder(session)
                 .setCatalogSessionProperty("iceberg", TARGET_SPLIT_SIZE, "1")
                 .build();
-        assertEquals(getSplitsForSql(minSplitSession, selectQuery), maxSplits);
+        assertEquals(getSplitsForSql(minSplitSession, selectQuery).size(), maxSplits);
         assertQuerySucceeds("DROP TABLE test_split_size");
     }
 
+    @Test
+    public void testSoftAffinitySchedulingSectionConfig()
+    {
+        Session maxIdentifiers = Session.builder(getSession())
+                .setCatalogSessionProperty("iceberg", AFFINITY_SCHEDULING_FILE_SECTION_SIZE, "1B")
+                .setCatalogSessionProperty("iceberg", TARGET_SPLIT_SIZE, "1")
```

Comment:
Should we make `TARGET_SPLIT_SIZE` a `DataSize` instead of refering to bytes ? We do this for `MAX_SPLIT_SIZE`. Can punt this to a new PR

Reply:
@aaneja Good catch!
@ZacBlanco Could you please address Anant's comment in this PR? I'll merge once it's done.

Reply:
This was done intentionally because the Iceberg library expects this configuration in bytes directly on the "read.target.split-size" table property. Iceberg has no notion of "DataSize" or unit-based shorthand for byte-typed values. In order to make setting the values for the table and session property consistent for the target split size I opted to not make this a DataSize-valued property.

Reply:
Spoke offline on this. We may rename TARGET_SPLIT_SIZE to TARGET_SPLIT_SIZE_BYTES. This is, of course, orthogonal to this PR

Reply:
PR to change the name: https://github.com/prestodb/presto/pull/24618